src/main/java/pl/sda/weather/controllers/ControllersLocation.java

```diff
@@ -1,9 +1,108 @@
 package pl.sda.weather.controllers;
 
 
+import pl.sda.weather.model.LocationModel;
+import pl.sda.weather.model.Weather;
+import pl.sda.weather.service.ILocationService;
+import pl.sda.weather.service.impl.LocationServiceImpl;
+import pl.sda.weather.service.impl.ReadWeathersImpl;
+import pl.sda.weather.validation.LocationValidation;
+
+import javax.xml.validation.ValidatorHandler;
+import java.util.List;
+import java.util.Map;
+import java.util.Scanner;
+import java.util.UUID;
+
+import static pl.sda.weather.Gui.showMainMenu;
+
 public class ControllersLocation {
+    ILocationService locationService = new LocationServiceImpl();
+    ReadWeathersImpl readWeathers = new ReadWeathersImpl();
+
+    private static final Scanner scanner = new Scanner(System.in);
+
+    private static LocationValidation validation = new LocationValidation();
+
+
+    public void addLocation() {
+
+        String idUUID = String.valueOf(UUID.randomUUID());
+
+       String longitudeAndLatitude;
+        String longitudeS;
+        String longitudeN;
+        String latitudeW;
+        String latitudeE;
+       do {
```

Comment:
would be greate to have terminal logic separated (MODEL_VIEW_CONTROLLER pattern) -> right now you have VIEW logic in CONTROLLER